firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -386,6 +386,16 @@ class BrowserViewController: UIViewController,
             } else {
                 navigationToolbarContainer.isHidden = true
             }
+
+            if let toolbarState = store.state.screenState(ToolbarState.self, for: .toolbar, window: windowUUID),
+               toolbarState.isShowingTopTabs != showTopTabs {
+                let action = ToolbarMiddlewareAction(
+                    isShowingTopTabs: showTopTabs,
+                    windowUUID: windowUUID,
+                    actionType: ToolbarMiddlewareActionType.traitCollectionDidChange
+                )
+                store.dispatch(action)
+            }
         } else {
```

Comment:
nit: maybe add private func that describes more why we are sending this call when `isShowingTopTabs != showTopTabs`

Reply:
discussed online, will wait for thursday meeting to finalize next steps